rfcs/rfc-manage-staff-role-and-account.md

```diff
@@ -0,0 +1,75 @@
+- Start Date: (2022-07-23)
+- Target Major Version: N/A
+- Reference Issues: (fill in existing related issues, if any)
+- Entity Issue: https://github.com/bcgov/entity/issues/12226
+- Implementation PR: None
+
+
+# Summary
+
+SBC staff need to access the staff dashboard with limited privileges to support SP/GP filing/alteration.
+
+This is an RFC draft to discuss more suitable approaches 
+
+# Basic example
+
+If the proposal involves a new or changed API, component, etc., include a basic code example.
+Omit this section if it's not applicable.
+
+# Motivation
+SBC staff want to access the staff dashboard to search for SP/GP filing for clients, They may have to register SP/GP companies on behalf of the client
+
+# Solution Options
+## 1) Use a new role for sbc staff from keycloak
+
+### Summary
+Keycloak Roles are currently used across the application to determine who can access the staff dashboard.
+We can create a new keycloak role for sbc-staff and in code, we can use this group to give permission for required components/pages.
+In this way, sbc-staff can do alterations for the organization on behalf of the user, and sbc-staff can use their GOVM account to create new SP/GP for Users
+We can create a new GOVM account and add all BC Registry staff into that GOVM account where our staff can access the products as sbc-staffs do.
+
+### Detailed design
+1. Have to create a new role in keycloak for sbc-staff and
+2. In code we have to check at all places where we need to give permission for the new role, like staffed
+3. In addition to account (where they are part), we can show the staff dashboard in the menu, where they can switch between staff dashboard and account 
+4. When sbc-staff/Registry staff access via staff dashboard, they can access all the accounts in our system (behalf of the user)
+5.  sbc-staff/Registry staff can use their  GOVM account to create new organization/filing etc .. on behalf of clients and they can manage it.
+6. sbc-staff/Registry staff  can use their  GOVM account  to access products like PPR/Name requests to help the clients
+7. As in existing we can have multiple staff groups to limit permission (search only/view/edit permissions)
+
+
+This approach will have a high dev impact on sbc-auth and pay-api, less dev impact on and all partner applications (can directly check user role from keycloak tokens)
+
+
+## 2) Bring all Registry staff into one account and connect accounts using account delegation
+
+### Summary
+Currently, the Registry staff is not part of any accounts, and the Registry staff's role is being verified directly from keycloak. As a part of this solution, all users will be under at least one account and Registry staff will be deligated to all accounts according to access permissions. Then we will handle granular permission for staff from sbc-auth like end-users.
+The same will apply to sbc-staff also. Sbc-staff will delegate access to accounts which they want to work on behalf of the client
```

Comment:
So under this model of delegating accounts, will it be easy to delegate all currently existing businesses and future businesses to Registry staff users?

I'm not familiar with the business needs for delegation.  Is there really a need for us to be able to delegate specific businesses to staff users?